earth_enterprise/docker/README.md

```diff
@@ -29,166 +29,217 @@ service docker start
 Now you can use Docker as the root user.
 
 
-## Pulling an Image from Docker Cloud
+## Building the Images
 
-You can get a pre-built image from a Docker registry.
+Before building Open GEE Docker images, make sure you have, at least, about
+8 GB of free disk space.
 
-```BASH
-docker pull thermopylae/opengee-experimental
-```
+### Image Configuration
 
+There are a number of option of how you can build and run Open GEE Docker
+images: 
 
-## Running the Container
+    * Platform to build on,
+    * Source code origin: from the main repository, or a fork, from the
+      master, or another branch, or from the current repository clone on your
+      disk;
+    * Port to forward HTTP connections to;
+    * Custom image to run, e.g., containing extra data like tutorial
+      resources, or custom built globes.
 
-### Running the Open GEE Servers
+These options are controlled by environment variables that are passed to the
+build scripts.  Since, it may become inconvenient to pass more than a couple
+of values on the command line, you could put the variable definitions in a
+shell script file instead, and source it before each command you execute.
 
-#### Running Open GEE Server with Persistent Globe and Map Databases
+Have a look at the files under
+`earthenterprise/earth_enterprise/docker/config` for examples.
 
-In order to be able to build and serve Earth databases that persist between 
-instantiations of the Docker image, you need a persistent assets directory 
-outside of the Docker container.  Another alternative is to run a container
-with a temporary assets directory, and commit the running to container to a
-new image after you've build your assets.
 
-To run the Docker container using a persistent globe assets directory outside 
-the container, use the following script.  Set the `HOST_GEVOL_PATH`
-environment variable to the path outside the container where you want to store
-your persistent assets.
+### Building a Build-environment Image
+
+Before being able to compile and build Open GEE in a Docker image, you need a
+prerequisite Docker image containing a build environment for the platform you
+want to build Open GEE on.
+
+The build environment image is reused every time you rebuild Open GEE.  This
+should save you time during normal development for targetting another
+platform, since only the steps of transferring the modified source code, and
+running the Open GEE build would need to be repeated.
+
+You can build a build-environment image with a command like this:
 
 ```BASH
-HOST_GEVOL_PATH=/persistent/gevol/ ./bin/start-docker-image.sh
+( . config/rhel-7.sh && ./bin/build-build-env-image.sh )
```

Comment:
Minor: I believe I have it figured out, but can you add a brief explanation on what the config/* files do for you?

Reply:
They allow to avoid indecently long command lines.

There's a subsection called _Image Configuration_ above that tries to explain that.